federationtypes.go

```diff
@@ -370,8 +370,14 @@ func (r RespState) Events() ([]Event, error) {
 	return result, nil
 }
 
+// MissingAuthEventHandler is a function signature which can be provided to
+// RespState.Check(), RespSendJoin.Check() and checkAllowedByAuthEvents. If
+// an auth event comes up missing during the checks, the MissingAuthEventHandler
+// will be called (if provided) to try and retrieve the missing event.
+type MissingAuthEventHandler func(eventID string, roomVersion RoomVersion) (*Event, error)
```

Comment:
Tempted to just use https://github.com/matrix-org/gomatrixserverlib/blob/master/authchain.go#L9

Reply:
That's not a bad idea actually. Let me take a look.